### PR TITLE
Added docker-service example to show how googlecloudspannerreceiver can work along with prometheus exporter and grafana dashboards

### PR DESCRIPTION
**Pull request by architjugran**
This example would help our receiver's users easily understand how to configure not only the receiver but also how to tie it together with other components.

### CODE REVIEW

**Comment by mx-psi** [comment]
@ydrozhdzhal @asukhyy @khospodarysko @architjugran can any of you review as codeowners?

**Review by ydrozhdzhal — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - `datadogexporter`: Add http.status_code tag to trace stats (#6889)
 - `tanzuobservabilityexporter`: Support delta histograms (#6897)
 - `mysqlreceiver`: Add the receiver to available components (#7078)
+- `googlecloudspannerreceiver`: Add an example to demonstrate usage
```
Comment (mx-psi) [inline CHANGELOG.md:13]: Please add the PR number

**Review by jpkrohling — COMMENTED** [review]
I'm a fan of end-to-end instructions to users and I love this aspect of this PR. But I've been bitten by outdated examples in the past. How can we ensure that the examples are still going to work in a year's time? Perhaps a blog post would be more suitable, as it's clear by the publication date that the contents might be outdated and not work anymore? You can then add a link here to the blog post.

File: receiver/googlecloudspannerreceiver/examples/googlecloudspanner-local/grafana/provisioning/datasources/cloud-monitoring-datasource.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    type: stackdriver
+    access: proxy
+    jsonData:
+      tokenUri: https://oauth2.googleapis.com/token
+      clientEmail: <YOUR SERVICE-ACCOUNT EMAIL> 
+      authenticationType: jwt
+      defaultProject: <YOUR SPANNER PROJECT NAME>
+    secureJsonData:
+      privateKey: |
+        <YOUR SERVICE-ACCOUNT PRIVATE KEY BELOW. SAMPLE SHOWN>
```
[the remark this answers is not in the document]
Comment (architjugran) [reply receiver/googlecloudspannerreceiver/examples/googlecloudspanner-local/grafana/provisioning/datasources/cloud-monitoring-datasource.yml:14]: Yes, this is a random secret key just added as a placeholder

**Comment by architjugran** [comment]
> I'm a fan of end-to-end instructions to users and I love this aspect of this PR. But I've been bitten by outdated examples in the past. How can we ensure that the examples are still going to work in a year's time? Perhaps a blog post would be more suitable, as it's clear by the publication date that the contents might be outdated and not work anymore? You can then add a link here to the blog post.

@jpkrohling , Yes - We will be releasing a blog soon and adding the link to this example there. Once that blog is released, I could add the link to the blog here too!

**Comment by jpkrohling** [comment]
Alright, I think the blog post would serve to replace this PR, instead of duplicating it. What do you think?

**Comment by architjugran** [comment]
> Alright, I think the blog post would serve to replace this PR, instead of duplicating it. What do you think?

No, the blog post would be about our receiver and would be linking to the example here -- explaining that customers can use this example for reference. 

Both the blog and this PR are needed, as this PR serves as storing our example code/configurations publicly.

**Comment by jpkrohling** [comment]
> Both the blog and this PR are needed, as this PR serves as storing our example code/configurations publicly.

You can have it as part of another repository, built specifically for this. In any case, it might be good to get other people's opinions. 

@open-telemetry/collector-contrib-approvers, how do you feel about having e2e examples as part of individual components?

**Comment by architjugran** [comment]
> > Both the blog and this PR are needed, as this PR serves as storing our example code/configurations publicly.
> 
> You can have it as part of another repository, built specifically for this. In any case, it might be good to get other people's opinions.
> 

What is the name of this other repository? Actually I am in a bit of a hurry with this PR, since our PM wants to be able to link to the example soon because of some internal deadlines. So would like to get this code pushed in, in this or the above other repo you are talking about.

I figured I could add my example in this repository itself after seeing a similar example already present at https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/simpleprometheusreceiver/examples/federation .

**Comment by mx-psi** [comment]
> > Both the blog and this PR are needed, as this PR serves as storing our example code/configurations publicly.
> 
> You can have it as part of another repository, built specifically for this. In any case, it might be good to get other people's opinions.
> 
> @open-telemetry/collector-contrib-approvers, how do you feel about having e2e examples as part of individual components?

I think it's a fairly common practice to have example OpenTelemetry Collector configuration files that showcase how to configure a given component, but it's not common to have full end to end examples. Keeping the example Collector configuration files up to date is doable by maintainers and contributors (and we can build tools to automate this if needed), but it's much harder to keep things that are not Collector config in a good state. I would also lean towards moving this to a separate repository/blog post.

**Comment by Aneurysm9** [comment]
> By "other repository", you mean my own new repo, right? Yeah we had considered that but thought that this should be the better place
> 
> Actually I am in a bit of a hurry with this PR, since our PM wants to be able to link to the example soon because of some internal deadlines.

If you have a deadline I suggest creating a repository under your control to house this.  We cannot guarantee that this PR will be accepted by your deadline.  We can always continue the discussion of whether this is an appropriate location once the deadline pressure is not a factor.

**Review by asukhyy — APPROVED** [review]

**Review by asukhyy — APPROVED** [review]

**Pull request closed without merging**